openpype/hosts/maya/api/lib.py

```diff
@@ -3446,3 +3446,17 @@ def dgcontext(mtime):
 def get_attribute_input(attr):
     connections = cmds.listConnections(attr, plugs=True, destination=False)
     return connections[0] if connections else None
+
+
+def write_xgen_file(data, filepath):
+    lines = []
+    with open(filepath, "r") as f:
+        for line in [line.rstrip() for line in f]:
+            for key, value in data.items():
+                if line.startswith("\t{}".format(key)):
+                    line = "\t{}\t\t{}".format(key, value)
```

Comment:
There are two instances of `xgDataPath` and `xgProjectPath` which needs changing. Ideally we should be probably parsing the Xgen file schema, but time does not allow for this.

Reply:
Ok, if there are more instances to change - I'd recommend this:
```suggestion
        for line in f:
            for key, value in data.items():
                if line.startswith("\t{}".format(key)):
                    line = "\t{}\t\t{}".format(key, value)
```

I couldn't find any logic why the lines needed to have `.rstrip()` called on them since it didn't seem you did anything with the end of the line to begin with. Unless you explicitly didn't want to write those end line values back into the file for a reason?

Reply:
Actually, NO! My previous suggestion was accurate. There might be more instances in the file, but the same line would never start with multiple keys - no? :) 

So it should still break out as soon as it did find a match and resolved it.

Reply:
We should not break out cause the lines are read to be, potentially modified and, written back in?

Reply:
We're not breaking out of the outer for loop, just out of the inner for loop where we check each key in data. As soon as we've found one we've already defined the `line =` value and thus can break back into the outer loop which appends it into the `lines`

Reply:
Cool, redundant now with this?
https://github.com/ynput/OpenPype/pull/4256#discussion_r1091851312

Reply:
I'd say so, yes - because it removes the for loop used here.